src/charm.py

```diff
@@ -29,12 +30,39 @@
 KUBE_CONFIG_PATH = "/home/ubuntu/.kube/config"
 
 
-def determine_arch() -> str:
-    """Dpkg wrapper to surface the architecture we are tied to."""
-    cmd = ["dpkg", "--print-architecture"]
-    output = subprocess.check_output(cmd).decode("utf-8")
+class KubeConfigResourceManager:
+    """Manage the kubeconfig resource."""
 
-    return output.rstrip()
+    def __init__(self, model: ops.model.Model):
+        self.kube_config_path = Path(KUBE_CONFIG_PATH)
+
+        try:
+            self.resource = model.resources.fetch("kubeconfig")
+        except NameError:
+            self.resource = None
```

Comment:
the integration tests failed with this:

```
2024-04-17 19:58:12 ERROR unit.kubernetes-e2e/0.juju-log server.go:325 certificates:18: Uncaught exception while in charm code:
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-kubernetes-e2e-0/charm/venv/ops/model.py", line 3022, in _run
    result = subprocess.run(args, **kwargs)  # type: ignore
  File "/usr/lib/python3.10/subprocess.py", line 526, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '('/var/lib/juju/tools/unit-kubernetes-e2e-0/resource-get', 'kubeconfig')' returned non-zero exit status 1.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-kubernetes-e2e-0/charm/./src/charm.py", line 199, in <module>
    ops.main(KubernetesE2ECharm)  # type: ignore
  File "/var/lib/juju/agents/unit-kubernetes-e2e-0/charm/venv/ops/main.py", line 555, in __call__
    return main(charm_class, use_juju_for_storage=use_juju_for_storage)
  File "/var/lib/juju/agents/unit-kubernetes-e2e-0/charm/venv/ops/main.py", line 544, in main
    manager.run()
  File "/var/lib/juju/agents/unit-kubernetes-e2e-0/charm/venv/ops/main.py", line 520, in run
    self._emit()
  File "/var/lib/juju/agents/unit-kubernetes-e2e-0/charm/venv/ops/main.py", line 509, in _emit
    _emit_charm_event(self.charm, self.dispatcher.event_name)
  File "/var/lib/juju/agents/unit-kubernetes-e2e-0/charm/venv/ops/main.py", line 143, in _emit_charm_event
    event_to_emit.emit(*args, **kwargs)
  File "/var/lib/juju/agents/unit-kubernetes-e2e-0/charm/venv/ops/framework.py", line 352, in emit
    framework._emit(event)
  File "/var/lib/juju/agents/unit-kubernetes-e2e-0/charm/venv/ops/framework.py", line 851, in _emit
    self._reemit(event_path)
  File "/var/lib/juju/agents/unit-kubernetes-e2e-0/charm/venv/ops/framework.py", line 941, in _reemit
    custom_handler(event)
  File "/var/lib/juju/agents/unit-kubernetes-e2e-0/charm/./src/charm.py", line 131, in _setup_environment
    kubeconfig_resource_manager = KubeConfigResourceManager(self.model)
  File "/var/lib/juju/agents/unit-kubernetes-e2e-0/charm/./src/charm.py", line 40, in __init__
    self.resource = model.resources.fetch("kubeconfig")
  File "/var/lib/juju/agents/unit-kubernetes-e2e-0/charm/venv/ops/model.py", line 1902, in fetch
    self._paths[name] = Path(self._backend.resource_get(name))
  File "/var/lib/juju/agents/unit-kubernetes-e2e-0/charm/venv/ops/model.py", line 3151, in resource_get
    out = self._run('resource-get', resource_name, return_output=True)
  File "/var/lib/juju/agents/unit-kubernetes-e2e-0/charm/venv/ops/model.py", line 3024, in _run
    raise ModelError(e.stderr) from e
ops.model.ModelError: ERROR could not download resource: HTTP request failed: Get https://10.246.153.64:17070/model/88aea862-a04f-43ee-81b4-50ee14f510d0/units/unit-kubernetes-e2e-0/resources/kubeconfig: while getting resource from charmhub: resource "kubeconfig" not found
```

looks like you need to catch `ops.ModelError`